packages/mongodb-explain-compat/lib/index.js

```diff
@@ -72,18 +72,56 @@ function mapStages(queryPlan, sbeExecutionStages) {
   });
 }
 
-module.exports = function(explain) {
-  explain = JSON.parse(JSON.stringify(explain));
-  if ((explain.explainVersion || 0) < 2) {
+function isAggregationExplain(explain) {
+  return !!explain.stages;
+}
+
+/**
+ * https://wiki.corp.mongodb.com/display/QUERY/Explain+Notes
+ */
+function mapAggregationExplain(explain) {
+  // In aggregations for SBE, the cursor based stage is always first.
+  const firstStage = explain.stages[0]; // {$cursor: {queryPlanner: {}, executionStats: {} }}
+  const firstStageKey = Object.keys(firstStage)[0]; // $cursor
+  if (!firstStageKey.match(/cursor/)) {
     return explain;
   }
-  delete explain.explainVersion;
+  const winningPlan =
+    firstStage[firstStageKey].queryPlanner.winningPlan.queryPlan;
+  firstStage[firstStageKey].queryPlanner.winningPlan = winningPlan;
+
+  firstStage[firstStageKey].executionStats.executionStages = mapStages(
+    winningPlan,
+    firstStage[firstStageKey].executionStats.executionStages
+  );
+  firstStage[firstStageKey].queryPlanner.plannerVersion = 1;
+  explain.stages[0] = firstStage;
```

Comment:
I think this line is a no-op, since `firstStage` is initialized to be `explain.stages[0]`.